pyaedt/hfss.py

```diff
@@ -2479,13 +2489,38 @@ def assign_primary(self, face, u_start, u_end, reverse_v=False, coord_name="Glob
         return self._create_boundary(primary_name, props, "Primary")
 
     def _create_pec_cap(self, sheet_name, obj_name, pecthick):
-        # TODO check method
-        obj = self.modeler[sheet_name].clone()
+        """Create a PEC object to back a wave port.
+
+
+         Parameters
+         ----------
+         sheet_name : str
+             Name of the sheet object touching the port surface.
+         obj_name : str
+             Name of the 3D object touching the port surface.
+         pecthick : float
+             Thickness of the PEC cap
+
+         Returns
+         -------
+        bool
+            ``True`` when successful, ``False`` when failed.
+
+        """
+        if isinstance(sheet_name, str) and isinstance(obj_name, cad.elements3d.FacePrimitive):
+            obj = obj_name.create_object()  # Create face object of type cad.object3d.Object3d from FacePrimitive
+            oname = obj_name._object3d.name
+            bounding1 = self.modeler[oname].bounding_box
+        else:
+            obj = self.modeler[sheet_name].clone()
+            bounding1 = self.modeler[obj_name].bounding_box
         out_obj = self.modeler.thicken_sheet(obj, pecthick, False)
         bounding2 = out_obj.bounding_box
-        bounding1 = self.modeler[obj_name].bounding_box
         tol = 1e-9
         i = 0
+
+        # Check that the pec cap is internal by comparing the bounding box
+        # of the cap with the bounding box of obj_name.
```

Comment:
Very useful for maintenance when we have those explainations. Thanks @dcrawforAtAnsys !